### DISCUSSION TITLE
Enable Tray Output on some but not all monitors

### DISCUSSION THREAD

**Discussion by tparrotte-cnyric** [i3/i3#5995] [Q&A] [answered]
My work laptop has three monitors when connected to the dock:
- eDP-1 1366x768 (the laptop screen - left)
- DP-1-2 1920x1080 (my main monitor - center)
- DP-1-3 1080x1920 (portrait mode -right)

I want my system tray to appear on eDP-1 and DP-1-2 at all times, but not on DP-1-3. (Even better would be always on DP-1-2 only while the computer is docked and only on eDP-1 when it is undocked.)

The `bar` section of my config file looks like this:

    bar {
        position bottom
        status_command i3status
        tray_output DP-1-2
        tray_output eDP-1
        colors {
            # bar background color
            background $bg
            # text color used for blocks that do not have a color specified.
            statusline $yellow
            # workspaces         section            
            #                    border  backgr. text
            focused_workspace    $aqua $aqua $darkgray
            inactive_workspace   $darkgray $darkgray $yellow
            active_workspace     $darkgray $darkgray $yellow
            urgent_workspace     $red $red $bg
        }
    }

With this configuration, the tray only appears on DP-1-2. I get an i3 error if I try to configure the tray_output as `tray_output DP-1-2 eDP-1`.

The i3 manual says this under the `tray_output` section:

> You can use multiple tray_output directives in your config to specify a list of outputs on which you want the tray to appear. The > first available output in that list as defined by the order of the directives will be used for the tray output.

From that, it seems to me like I've configured this correctly, but it doesn't work the way I expect. Can anyone point me in the right direction? Thanks!

**Comment by okraits** [comment, marked as answer]
In your quote from the docs the word `first` is the important one. The tray will be shown on the first available output from that list defined in your configuration, and only on that output.
